### PR TITLE
Make convolution_nd function work with ndarray.

### PR DESCRIPTION
**Pull request by takagi**
Related to #1687 .

This PR makes `convolution_nd` function accept `ndarray` as its `x` argument. Its docstring is modified in #1687 .

### CODE REVIEW

**Comment by zori** [comment]
Could you also modify the tests in functions_tests and links_tests to test with ndarray input? Current test code always wraps with `chainer.Variable`.

**Comment by takagi** [comment]
I added the test for the case that `numpy.ndarray` and `cupy.ndarray` are supplied to `convolution_nd` function, would check it again?

File: tests/chainer_tests/functions_tests/connection_tests/test_convolution_nd.py [hunk 1]
```diff
@@ -231,4 +231,39 @@ def test_call_cudnn_backward(self):
             y.backward()
             self.assertEqual(func.called, self.expect)
 
+
+class TestConvolutionNDNdarraySupplied(unittest.TestCase):
```
Comment (zori) [inline tests/chainer_tests/functions_tests/connection_tests/test_convolution_nd.py]: To improve readability, consider renaming to `TestConvolutionNDarraySupplied` (without the second **nd**)

File: tests/chainer_tests/functions_tests/connection_tests/test_convolution_nd.py [hunk 2]
```diff
@@ -231,4 +231,39 @@ def test_call_cudnn_backward(self):
...
+        out_channels = 2
+        dtype = numpy.float32
+
+        x_shape = (N, in_channels, 3, 3, 3)
+        self.x_data = numpy.random.uniform(-1, 1, x_shape).astype(dtype)
+        W_shape = (out_channels, in_channels, 1, 1, 1)
+        self.W_data = numpy.random.uniform(-1, 1, W_shape).astype(dtype)
+        self.b_data = numpy.random.uniform(-1, 1, out_channels).astype(dtype)
+
+    def check_ndarray_supplied(self, x_ary, W_ary, b_ary):
```
Comment (zori) [inline tests/chainer_tests/functions_tests/connection_tests/test_convolution_nd.py]: ditto `check_array_supplied`

File: tests/chainer_tests/functions_tests/connection_tests/test_convolution_nd.py [hunk 3]
```diff
@@ -231,4 +231,39 @@ def test_call_cudnn_backward(self):
...
+        y_ary = functions.convolution_nd(x_ary, W_ary, b_ary)
+
+        x_var = chainer.Variable(x_ary)
+        W_var = chainer.Variable(W_ary)
+        b_var = chainer.Variable(b_ary)
+        y_var = functions.convolution_nd(x_var, W_var, b_var)
+
+        testing.assert_allclose(y_ary.data, y_var.data)
+
+    def test_ndarray_supplied_cpu(self):
```
Comment (zori) [inline tests/chainer_tests/functions_tests/connection_tests/test_convolution_nd.py]: ditto

File: tests/chainer_tests/functions_tests/connection_tests/test_convolution_nd.py [hunk 4]
```diff
@@ -231,4 +231,39 @@ def test_call_cudnn_backward(self):
...
+        self.check_ndarray_supplied(self.x_data, self.W_data, self.b_data)
+
+    @attr.gpu
+    def test_ndarray_supplied_gpu(self):
```
Comment (zori) [inline tests/chainer_tests/functions_tests/connection_tests/test_convolution_nd.py]: ditto

File: tests/chainer_tests/functions_tests/connection_tests/test_convolution_nd.py (hunk 1 above)
Comment (takagi) [inline tests/chainer_tests/functions_tests/connection_tests/test_convolution_nd.py]: Fixed.

File: tests/chainer_tests/functions_tests/connection_tests/test_convolution_nd.py (hunk 2 above)
Comment (takagi) [inline tests/chainer_tests/functions_tests/connection_tests/test_convolution_nd.py]: Fixed.

File: tests/chainer_tests/functions_tests/connection_tests/test_convolution_nd.py (hunk 3 above)
Comment (takagi) [inline tests/chainer_tests/functions_tests/connection_tests/test_convolution_nd.py]: Fixed.

**Comment by takagi** [comment]
Thanks, I fixed them.

**Comment by zori** [comment]
Thank you!

**Pull request merged**